### PR TITLE
filter node list by node status

### LINKED ISSUE #1072: Filter preload ended nodes only in node list

### PR DESCRIPTION
**Pull request by ipdae**
- resolve #1072 
- Fix `NodeInfo` select. using ipc communication instead of config variables.
- refactoring app initialize for waiting node list.

before
```
13:28:48.231 › Executing the remote headless server:
  a0d01f897e0a2434798e8b4607ac32ea-994288696.us-east-2.elb.amazonaws.com/31238
1:30
Request URL: http://adb1932b4da92426abd7116c65875faa-1809698636.us-east-2.elb.amazonaws.com/graphql
```

after
```
18:47:58.718 › Executing the remote headless server:
  aba105ce4868747839ab806175be8b37-13503424.us-east-2.elb.amazonaws.com/31238
Request URL: http://aba105ce4868747839ab806175be8b37-13503424.us-east-2.elb.amazonaws.com/graphql

```

### CODE REVIEW

File: src/renderer/App.tsx [hunk 1]
```diff
@@ -117,6 +75,59 @@ function App() {
...
+    wsLink,
+    httpLink
+  );
+
+  const link = ApolloLink.from([new RetryLink(), apiLink]);
+
+  const client = new ApolloClient({
+    link: link,
+    cache: new InMemoryCache(),
+  });
```
Comment (BasixKOR) [inline src/renderer/App.tsx:115]: These changes will cause a creation of a new Apollo Client on every render, which is highly undesirable. I know we gotta get URL via IPC first but still we gotta find a better way.

File: src/collection/App.tsx [hunk 2]
```diff
@@ -63,19 +39,49 @@ function createFile() {
...
+        definition.operation === "subscription"
+      );
+    },
+    wsLink,
+    httpLink
+  );
+
+  const link = ApolloLink.from([new RetryLink(), apiLink]);
+
+  const client = new ApolloClient({
```
Comment (BasixKOR) [inline src/collection/App.tsx:80]: See comments on `src/renderer/App.tsx`.

File: src/main/main.ts [hunk 3]
```diff
@@ -551,9 +567,10 @@ function initializeIpc() {
     utils.deleteBlockchainStoreSync(getBlockChainStorePath());
     if (rerun) {
       if (useRemoteHeadless) {
-        initializeRemoteHeadless();
+        console.log("main clear cache call initializeRemoteHeadless");
+        await initializeRemoteHeadless();
       } else {
-        initializeHeadless();
+        await initializeHeadless();
```
Comment (BasixKOR) [inline src/main/main.ts:573]: Just a curiosity, what does this do?

File: src/renderer/stores/game.ts [hunk 4]
```diff
@@ -43,26 +43,18 @@ export default class GameStore {
   }
 
   @action
-  startGame = (privateKey: string) => {
+  startGame = async (privateKey: string) => {
```
Comment (BasixKOR) [inline src/renderer/stores/game.ts:46]: I couldn't find a call to `startGame` using Promise or `await`. Is this change intended?

**Review by BasixKOR — CHANGES_REQUESTED** [review]
Thank you for your hard work. However, there are some changes that need further discussion IMO.

File: src/v2/utils/useStore.tsx [hunk 5]
```diff
@@ -6,7 +6,6 @@ import StandaloneStore from "../stores/standaloneStore";
 
 const stores = {
   account: new AccountStore(),
-  game: new GameStore(),
```
[the remark this answers is not in the document]
Comment (ipdae) [reply src/v2/utils/useStore.tsx:9]: I found it when change the GameStore constructor, there was no place currently used the stores.GameStore.

File: src/renderer/stores/game.ts (hunk 4 above)
Comment (ipdae) [reply src/renderer/stores/game.ts:46]: It seems previous work remain. I'll delete it.

File: src/main/main.ts (hunk 3 above)
Comment (ipdae) [reply src/main/main.ts:573]: It seems previous work remain. I'll delete it.

File: src/renderer/stores/game.ts (hunk 4 above)
Comment (ipdae) [reply src/renderer/stores/game.ts:46]: It seems previous work remain. I'll delete it.

File: src/main/main.ts [hunk 6]
```diff
@@ -103,6 +105,7 @@ const client = new NTPClient("time.google.com", 123, { timeout: 5000 });
 
 let remoteHeadless: RemoteHeadless;
 let useRemoteHeadless: boolean;
+let remoteNode: NodeInfo;
```
[the remark this answers is not in the document]
Comment (ipdae) [reply src/main/main.ts:108]: How can I move Apollo initialization?

[the remark this answers is not in the document]
Comment (BasixKOR) [reply src/main/main.ts:108]: Assuming we have `remoteNode` determined at the start of process and it doesn't change afterwards, we could make a quick handler like this in main process:

```ts
ipcMain.on("get remote node", event => event.returnValue = remoteNode);
```

and call them as sendSync like this: `ipcRenderer.sendSync("get remote node")`. Of course this has performance implications as well but since we are just returning same value and we wouldn't make a lot of calls to them I think it should be fine.

File: src/config.ts [hunk 7]
```diff
@@ -197,35 +187,58 @@ export class NodeInfo {
...
+      `http://${this.GraphqlServer()}`,
+      {
+        query: "query{nodeStatus{preloadEnded}}",
+      },
+      {
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    );
```
Comment (BasixKOR) [inline src/config.ts:202]: Just a note but we can reuse `graphql-request` module here since we already have it as a dependency.

File: src/collection/App.tsx [hunk 8]
```diff
@@ -41,47 +41,48 @@ const isFileExsist = getIsFileExsist();
 
 const App: React.FC = () => {
   const [agentAddress, setAgentAddress] = useState<string>("");
-  const [headlessUrl, setHeadlessUrl] = useState<string>("");
+  const [client, setClient] = useState<ApolloClient<any>>();
 
-  ipcRenderer.on("initialize collection window", (_, address, node) => {
+  ipcRenderer.on("initialize collection window", (_, address, headlessUrl) => {
```
Comment (BasixKOR) [inline src/collection/App.tsx:46]: Could we wrap this code in useEffect? I'll do that if you don't mind.

**Review by BasixKOR — APPROVED** [review]
Overall nice.
